processor/tailsamplingprocessor/internal/sampling/composite_test.go

```diff
@@ -0,0 +1,250 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package sampling
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+)
+
+type FakeTimeProvider struct {
+	second int64
+}
+
+func (f FakeTimeProvider) getCurSecond() int64 {
+	return f.second
+}
+
+var traceID = pdata.NewTraceID([16]byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x52, 0x96, 0x9A, 0x89, 0x55, 0x57, 0x1A, 0x3F})
+
+func createTrace() *TraceData {
+	trace := &TraceData{SpanCount: 1}
+	return trace
+}
+
+func TestCompositeEvaluatorNotSampled(t *testing.T) {
+
+	// Create 2 policies which do not match any trace
+	n1 := NewNumericAttributeFilter(zap.NewNop(), "tag", 0, 100)
+	n2 := NewNumericAttributeFilter(zap.NewNop(), "tag", 200, 300)
+	c := NewComposite(zap.NewNop(), 1000, []SubPolicyEvalParams{{n1, 100}, {n2, 100}}, FakeTimeProvider{})
+
+	trace := createTrace()
+
+	decision, err := c.Evaluate(traceID, trace)
+	if err != nil {
+		t.Fatalf("Failed to evaluate composite policy: %v", err)
+	}
+
+	// None of the numeric filters should match since input trace data does not contain
+	// the "tag", so the decision should be NotSampled.
+	expected := NotSampled
+	if decision != expected {
+		t.Fatalf("Incorrect decision by composite policy evaluator: expected %v, actual %v", expected, decision)
+	}
+}
+
+func TestCompositeEvaluatorSampled(t *testing.T) {
+
+	// Create 2 subpolicies. First results in 100% NotSampled, the second in 100% Sampled.
+	n1 := NewNumericAttributeFilter(zap.NewNop(), "tag", 0, 100)
+	n2 := NewAlwaysSample(zap.NewNop())
+	c := NewComposite(zap.NewNop(), 1000, []SubPolicyEvalParams{{n1, 100}, {n2, 100}}, FakeTimeProvider{})
+
+	trace := createTrace()
+
+	decision, err := c.Evaluate(traceID, trace)
+	if err != nil {
+		t.Fatalf("Failed to evaluate composite policy: %v", err)
+	}
+
+	// The second policy is AlwaysSample, so the decision should be Sampled.
+	expected := Sampled
+	if decision != expected {
+		t.Fatalf("Incorrect decision by composite policy evaluator: expected %v, actual %v", expected, decision)
+	}
+}
+
+func TestCompositeEvaluatorSampled_AlwaysSampled(t *testing.T) {
+
+	// Create 2 subpolicies. First results in 100% NotSampled, the second in 100% Sampled.
+	n1 := NewNumericAttributeFilter(zap.NewNop(), "tag", 0, 100)
+	n2 := NewAlwaysSample(zap.NewNop())
+	c := NewComposite(zap.NewNop(), 10, []SubPolicyEvalParams{{n1, 20}, {n2, 20}}, FakeTimeProvider{})
+
+	for i := 1; i <= 3; i++ {
```

Comment:
What is this test actually doing? What's the difference with the previous test? Why starting with 1 here?

Reply:
This test is same as before, just that we are checking the edgecase. I need to change 3 to 10. Since we have set maxTotalSpansPerSecond=10, we should be able to sample 10 traces. Set to 1 to make it readable1 to 10.